pkg/component/watch.go

```diff
@@ -255,7 +256,7 @@ func WatchAndPush(client *occlient.Client, out io.Writer, parameters WatchParame
 						glog.V(4).Infof("error removing watch for %s: %v", event.Name, e)
 					}
 					// append the file to list of deleted files
-					if !alreadyInChangedFiles && !matched {
+					if !alreadyInChangedFiles && !matched && event.Name != "" {
```

Comment:
when it is that `even.Name` can be empty?

Reply:
can mention this in the comment, so in future, we know when Name can be empty

Reply:
Sure

Reply:
Done